### DISCUSSION TITLE
Welcome to Sovol-SV06-firmware Discussions!

### DISCUSSION THREAD

**Discussion by bassamanator** [bassamanator/Sovol-SV06-firmware#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by shogun-fpv** [comment]
Hello Bassam. Thanks for putting this together.
Do you have some basic Slicer settings that are specific to your configuration?
The documentation makes reference of Andrew Ellis' guide, but wondering if you had specific Gcodes for your setup.

Thanks.

**Reply by bassamanator** [reply]
@shogun-fpv Not sure if I follow. All the macros included are listed under `Highlights`. The SuperSlicer config bundle already includes the start and end gcodes.

**Comment by dweenz** [comment]
Hi Bassam, Hope you are well. I know I can just add the BED_MESH_PROFILE LOAD=default to my gcode to load my mesh, but could you explain if you have time how to create the macro to add the above command with the m420 command in it's place I am on klipper and m420 does not work.
Cheers,

**Reply by bassamanator** [reply]
@dweenz My friend you missed the last step, see the 'adjust your slicer section'. All of these nitty gritty details are taken care of when using this config.

**Comment by LzLang** [comment]
Hello Bassam,
thanks for your work!

I'm new to Klipper and I have a question about `position_max `:
When I use the command `G1 X300 F3000` and `G1 Y300 F3000` I have about 3 mm on the X-Axis and 5 mm on the Y-Axis till it "hit's" the frame.
My question is now: should the `max_position` for X: 303 and for Y:305 because that the maximum movement, or is `max_position` only important for the overall print size?
(Same goes for the Z-Axis)

Thanks in advance!

**Comment by bassamanator** [comment]
@LzLang So `max_position` is not a thing, so I'll just ignore it.

From klipper docs:
```
position_max:
#   Maximum valid distance (in mm) the user may command the stepper to
#   move to. This parameter must be provided for the X, Y, and Z
#   steppers on cartesian style printers.
```
So if you can actually get to X303, then you can set 303mm as your X build plate size for example. I recommend to leave 2-3mm as a margin for error, in my case my filament sensor cable hangs around that area and I would never want it to get pinched. Make adjustments according to your needs.

**Comment by Aquaman7** [comment]
will this work on BigTreeTechs pad 7? I got it for review and wanted to use it on the sv06 plus because it prints so slow but I'm kind of new to klipper and not entirely sure I'm doing this right.

**Reply by bassamanator** [reply]
Yes it will. Here's a [video](https://youtu.be/zQZPUjVn3Vo) using my config and the pad 7, however, it's in German, but you can use the auto-translate subtitles feature.

**Reply by Manfredo1952** [reply]
Indeed. I am from Germany  and have used the video linked by Bassam to be convinced to by it for the SV06+ that I have.  Am using the Pad7 with my SV06+ quite happily.

**Reply by bassamanator** [reply]
@Manfredo1952 I was quite delighted when I came across that video, even though I couldn't understand any of it, and had to use the auto-translated subtitles. I'm glad my config has made it around the world.

And that `mpoxDE` is a super nice dude!

**Comment by milan360** [comment]
Hey Bassam,

I have an SV06 with an SKR-mini-e3-v3 board and recently tried your config.  The issue I'm having is once I load up and start a print job, and both the bed and hotend reach their temperature, both of the heaters turn off and the printer keeps the print going until the temperatures get too low and Klipper just cancels the print.

I am using Cura to slice, and have updated my gcode start and end to reflect what you have in your instructions: 

`
PRINT_START BED=[first_layer_bed_temperature] HOTEND={first_layer_temperature[initial_extruder]+extruder_temperature_offset[initial_extruder]} CHAMBER=[chamber_temperature]
PURGE_LINE
`
Is there something simple I'm missing?

**Reply by bassamanator** [reply]
The start gcode looks wrong to me, please see Adjust your slicer section. Also, for Cura, it's a 1 line start gcode, the `purge_line` must be on its own line, so 2 lines if using `purge_line`.

**Reply by milan360** [reply]
Hi, thanks for the reply.  I copied the code from your github directly.  And yes, the purge line is on a separate line, this may have been a typo on my part when I was pasting it into the discussion here.

<img width="376" alt="image" src="https://github.com/bassamanator/Sovol-SV06-firmware/assets/28825153/8eb4e833-652a-44b3-81ba-389f9da636dd">


<img width="868" alt="image" src="https://github.com/bassamanator/Sovol-SV06-firmware/assets/28825153/839209c2-3ba5-4cfc-b255-87b2fe1b1e16">

**Reply by bassamanator** [reply]
This is the correct start gcode for cura:
```
PRINT_START BED={material_bed_temperature_layer_0} HOTEND={material_print_temperature_layer_0} CHAMBER={build_volume_temperature}
```
Feel free to add the `purge_line` on a second line.

**Reply by bassamanator** [reply]
It says clearly that this is an **example**.

**Reply by milan360** [reply]
Understood.  Thank you for the reply and the help.

**Reply by bassamanator** [reply]
This whole Klipper switch is quite technical, it's crucial that the instructions are followed to a `T`, they are correct and have been tested thousands of times.

**Reply by bassamanator** [reply]
@milan360 Instructions have been tweaked based on your feedback. Thanks!

Feel free to report other issues.

**Comment by lrochfort** [comment]
Hi @bassamanator , thanks for the guide.

I've just installed Mainsail using MainsailOS.   It says to import mainsail.cfg into the printer.cfg.  I see you don't have that in your printer.cfg.

Should I add it?

**Reply by bassamanator** [reply]
I'm a MainsailOS user my self and I never had to do this. However, it's possible that this is a recent change. Please do a `[include .mainsail.cfg]` in your `printer.cfg`, somewhere near the top. Will look into this.

**Reply by bassamanator** [reply]
I just checked this with the latest version of MainsailOS. `mainsail.cfg` is **not required**. You've missed a step somewhere.

**Reply by lrochfort** [reply]
Thanks for looking.

I looked here: https://docs-os.mainsail.xyz/getting-started/first-boot

In the "Config" section it states:

We recommend checking mainsail.cfg and adapting it to your needs. Then open the printer.cfg and type in the editor [include mainsail.cfg] and click on "SAVE & RESTART".

**Reply by bassamanator** [reply]
Recommendation, not a requirement...

**Comment by chainsawsquirrel** [comment]
Im going nutz, the sv06 plus wont home properly, thinks the back of the bed is home position , Ive copied your stepper settings into the config.cfg still doing it .. is there another place home etc defined ?

**Comment by tayloredenterprises** [comment]
Hiya! 

Wondered if you could help me please. 

I have upgraded my sv06+ with a BTT skr e3 3.0 and Klipper. Using stock motors.

The first layer looks good however in subsequent layers, the extruder seems to be skipping steps / not rising as it should which results in the nozzle going into the previous layer. 

I’ve tried adapting your BTT / sv06 printer config, however it doesn’t fix the problem. Is there a different stepper/driver config required for BTT with sv06+? Particularly microsteps , rotation distance and current?

**Reply by bassamanator** [reply]
So it seems that everything is working? Simply pull in stepper currents from sv06-plus branch for the steppers. Also read warning about steppers on skr branch.

# Original wrote this but ignore it since you have everything working already

You need to carefully blend 2 branches together, sv06-plus, skr-mini-e3-v3.

Start off with skr-mini-e3-v3:
1. apply dimensions from sv06-plus
2. apply run_current from sv06-plus
3. handle filament sensor after you have everything working, so leave it commented out

**Comment by ChrSch84** [comment]
@bassamanator 

Hello first of all i love your klipper Guide :) Thanks for the hard work.

Now my Question. 

I found out that there is a Klipper Default Config for the S06 Plus and there a few changes to your Config.

Found it on the Github for Klipper.
[https://github.com/Klipper3d/klipper/blob/master/config/printer-sovol-sv06-plus-2023.cfg](url)

i Compared the two files and this is a small snapshot of it.

Left is your Config.
"
![image](https://github.com/bassamanator/Sovol-SV06-firmware/assets/30344592/7730caab-5c6c-4eb3-be30-e056cba76551)
"

Some Settings are faster with yours but not all.

Could you give us some Info why you use "this" settings over the "klipper" ones? Mybe Pro and Con?

Thank you

**Reply by bassamanator** [reply]
I added a second paragraph to the [Preface](https://github.com/bassamanator/Sovol-SV06-firmware?tab=readme-ov-file#preface) on the repo to answer such questions quite some time ago. Please read that, and further, know that the config found on the Klipper repo is not created by someone who writes code for the Klipper repo, or some 3d printing expert, and it is certainly not the "source of truth" (that would be Sovol's Marlin source code and now the Klipper config that comes with the Sovol Klipper screen).

Glad you found the repo useful. Please also see the [everything-sovol-sv06](https://github.com/bassamanator/everything-sovol-sv06) repo.

**Reply by ChrSch84** [reply]
Ok thank you.

Found the Klipper Config from the Klipper Screen.

But there is even more split of the settings.

Example:

"[printer]
kinematics: cartesian
max_velocity: 300
max_accel: 10000
max_accel_to_decel:5000
max_z_velocity: 15
max_z_accel: 200
square_corner_velocity:5.0"

10k vs 3k and other Settings like the stepper Motors. I know your settings are your settings... i get it but why is there so a big gap? 

Is it that sovol is "fuck it" give them speed over quality or is there something else?

We all here want to push our printers but at what cost? Hope you could shine a light on this. 

Thank you.

**Reply by bassamanator** [reply]
The properties that you're pointing to, max_velocity, max_accel, etc., are meaningless, as in, they will vary from printer to printer in the budget market. You need to adjust those to your needs (check the personal branch to see the settings that I use, for example). And yes, Sovol doesn't care about these details unfortunately, I'm not joking here, it's a bit of a sad truth, but this can be said about most manufacturer's (companies such as Bambu Lab being an exception). For example, I've encountered one or two users who can't push their printers past 3k/4k accel, I'm sure there are many more of them. 

You're overthinking this when you should be testing and adjusting these values to the capabilities of your particular unit.

**Comment by tjhayko** [comment]
I'm not sure if this is the correct place to ask this, but I'm trying to get my SV06plus working with a BTT Pad 7 based on bassmanator's instructions, but /dev/serial doesn't seem to exist :(

I've flashed CB1_Debian11_Klipper_kernel5.16_20240319.img.xz onto my SD card, so I think I'm on the latest OS for the BTT Pad 7 with CB1.

Any help or pointers would be appreciated.

**Reply by bassamanator** [reply]
Try the other method listed. Also try asking for help [here](https://github.com/bassamanator/Sovol-SV06-firmware/discussions/181).

**Reply by tjhayko** [reply]
Thanks for the quick reply.  The whole folder "/dev/serial" doesn't exist, so neither of the paths "/dev/serial/by-id/" or "/dev/serial/by-path/" exist.  

Is there some other way to get it connected?

**Reply by bassamanator** [reply]
What is the output from `lsusb` when the printer is connected and when it's not connected? Make sure it's on.

What is the output from `groups`?

**Reply by tjhayko** [reply]
Thanks again. 

The output from lsusb is the same (at least according to 'diff') regardless on whether the printer is connected or not:

```
Bus 008 Device 001: ID 1d6b:0001 Linux Foundation 1.1 root hub
Bus 004 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 007 Device 001: ID 1d6b:0001 Linux Foundation 1.1 root hub
Bus 003 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 006 Device 001: ID 1d6b:0001 Linux Foundation 1.1 root hub
Bus 002 Device 004: ID 05dc:c75c Lexar Media, Inc. JumpDrive V10
Bus 002 Device 003: ID 04d9:8030 Holtek Semiconductor, Inc. BTT-HDMI7
Bus 002 Device 002: ID 1a40:0101 Terminus Technology Inc. Hub
Bus 002 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 005 Device 001: ID 1d6b:0001 Linux Foundation 1.1 root hub
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub

```
I've tried multiple USB cables, and all seem to behave the same.  I tried using one of the cable with a Windows machine, and while there didn't seem to be a device that I noticed in Windows Device Manager. Windows did make it's "USB device connected" notification sound.  I also plugged in a Lexus JumpDrive into the Pad 7 to check to see if USB was working at all.  Seems like it is :)

Here's the output from groups:

`biqu tty disk dialout sudo audio video plugdev games users systemd-journal input netdev ssh`

**Reply by bassamanator** [reply]
*If* in fact the diff you did from `lsusb` is correct, then the klipper host, pad 7 in this case, is unable to detect the printer motherboard. I guess you could try flashing marlin, making sure it's up and running, then connecting it to the pad 7 to check to see if it is able to detect it. If the pad 7 still doesn't detect the printer, then you have some fundamental issue here. You can also try updating the pad 7 via `sudo apt update && sudo apt full-upgrade`.

**Reply by tjhayko** [reply]
I did an the apt update & ugprade when I setup the Pad 7.

I tried with Marlin as well, and the Pad 7 is still not able to detect the printer.

My next step is to try seeing what Windows detects when it connects to printer.  Perhaps Sovol has changed something.  I know the USB UART converters are something that often get changed in microcontrollers, so maybe that's what is happening here.  I'll post back when get some results from that.

**Reply by tjhayko** [reply]
So WIndows apparently knows that the USB device is connected, but it doesn't seem to know what it is :(

I've heard that the Sovol SV06 and SV06plus both use a CH340 UART, but there are number of CH340x chips, and I'm not sure if they use different drivers :(

I tried downloading and loading a Linux driver from here: https://github.com/juliagoda/CH341SER?tab=readme-ov-file#compatibility but that didn't seem to help either :(

**Reply by tjhayko** [reply]
Ugh, even though all of the USB cables I tried seem to work fine with an old mini USB phone, I finally tried picking up a new cable today, and now the Pad 7 seems to get a connection, or at least the output from dmesg seems to think so.  I'm seeing this in dmesg now:

```
[    6.796170] usbcore: registered new interface driver ch341
[    6.796339] usbserial: USB Serial support registered for ch341-uart
[    6.796464] ch341 2-1.3:1.0: ch341-uart converter detected
[    6.798615] usb 2-1.3: ch341-uart converter now attached to ttyUSB0
```

I put /dev/ttyUSB0 in my printer.cfg (in ~/printer_data) as follows:

```
[mcu]
# NOTE Obtain definition by "ls -l /dev/serial/by-path/" or "ls -l /dev/serial/by-id/"
serial: /dev/ttyUSB0
restart_method: command
```
 and I'm now getting the following error in my klippy.log which seems to repeat a few times:

```
mcu 'mcu': Starting serial connect
mcu 'mcu': Timeout on connect
mcu 'mcu': Wait for identify_response
Traceback (most recent call last):
  File "/home/biqu/klipper/klippy/serialhdl.py", line 68, in _get_identify_data
    params = self.send_with_response(msg, 'identify_response')
  File "/home/biqu/klipper/klippy/serialhdl.py", line 262, in send_with_response
    return src.get_response([cmd], self.default_cmd_queue)
  File "/home/biqu/klipper/klippy/serialhdl.py", line 319, in get_response
    self.serial.raw_send_wait_ack(cmds[-1], minclock, reqclock,
  File "/home/biqu/klipper/klippy/serialhdl.py", line 254, in raw_send_wait_ack
    self._error("Serial connection closed")
  File "/home/biqu/klipper/klippy/serialhdl.py", line 61, in _error
    raise error(self.warn_prefix + (msg % params))
serialhdl.error: mcu 'mcu': Serial connection closed
```

**Reply by bassamanator** [reply]
The path is incorrect. Use the `ls` command found in the `README` instead.
![Screenshot_20240902_144216](https://github.com/user-attachments/assets/f8500604-c582-4136-929f-01bf8b0d1b96)

**Reply by tjhayko** [reply]
Okay, I changed that, and I'm still getting the same error in my klippy.log :(

Is there any way to determine if the printer was correctly flashed with the klipper firmware?  That's a wild guess on my part, but now that the Pad appears to be able to recognize that the printer is connected.

I also found that the moonraker.conf had the klippy_uds_address set incorrectly to  /home/pi/printer_data/comms/klippy.sock, instead of /home/biqu/printer_data/comms/klippy.sock, so I fixed that.  Now at least I'm getting the error in the web UI rather than having to go to the logs.

**Reply by bassamanator** [reply]
> I also found that the moonraker.conf had the klippy_uds_address set **incorrectly** to /home/pi/printer_data/comms/klippy.sock ...

This is already clearly mentioned in the documentation, which you obviously haven't read, hence your struggle. It takes 9 minutes to read the entire page. I'm sure you've missed other things as well.

**Reply by tjhayko** [reply]
Thanks. I've read the page a couple of times, but I guess I unfortunately missed that part.

After reading again, I think the only step I've missed is disabling the 5V pin on the USB cable. Is the a possible cause of this type of problem? I've ordered a USB power blocker to try, but it won't be here for a few days

**Reply by tjhayko** [reply]
Apparently, microSD card #4 is the charm.  I reflashed again with yet another SD card.  I'm able to get connected it looks like mainsail is working or at least I'm able to home the extruder and move the head around now.

**Reply by bassamanator** [reply]
Great. Just run through the rest of the steps carefully, you're more likely to have a collision during the initial setup. Best of luck!
